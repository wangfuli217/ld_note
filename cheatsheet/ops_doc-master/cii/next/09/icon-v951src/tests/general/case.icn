#SRC: JCON

record rec(a)

procedure main(args)
   local L, s, x, r, c

   r := rec(45)
   c := create 1 | 2
   L := [&null, 0, 1, 2, 0.0, 1.0, 2.0, "", "0", "1", "2", '', '0', '1', '2',
      rec, main, foo, rec, rec(), r, c, []]
   put(L, L)

   every x := !L do {
      s := case x of {
        1:        "1"
	'1':      "'1'"
        1.0:      "1.0"
        "1":      "\"1\""
        &null:    "null"
        main:     "main"
        rec:	  "rec"
	rec():    "rec()"	# shouldn't ever match
	r:        "r"
	c:        "c"
	L:        "L"
        default:  "default"
        }
      write(right(s, 10), " : " , image(x))
      }
end

procedure foo()
end
