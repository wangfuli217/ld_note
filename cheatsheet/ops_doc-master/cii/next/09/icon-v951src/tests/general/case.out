      null : &null
   default : 0
         1 : 1
   default : 2
   default : 0.0
       1.0 : 1.0
   default : 2.0
   default : ""
   default : "0"
       "1" : "1"
   default : "2"
   default : ''
   default : '0'
       '1' : '1'
   default : '2'
       rec : record constructor rec
      main : procedure main
   default : procedure foo
       rec : record constructor rec
   default : record rec_2(1)
         r : record rec_1(1)
         c : co-expression_2(0)
   default : list_2(0)
         L : list_3(24)
